src/lib.rs

```diff
@@ -416,40 +390,254 @@ fn load() {
+    let group_member: GroupMemberOwned = group_member.into();
+    let event = ControllerEvent::RTAPISquadUpdate(state, group_member);
+    let _event_send = sender.try_send(event);
+}
+
+fn receive_squad_update<'u>(update: impl IntoIterator<Item = &'u UserInfo>) {
+    let update: Vec<_> = update.into_iter()
+        .map(|x| unsafe { ptr::read(x) }.into())
+        .collect();
+    let sender = CONTROLLER_SENDER.get().unwrap();
+    let event = ControllerEvent::ExtrasSquadUpdate(update);
+    let _event_send = sender.try_send(event);
+}
+
+fn render_overlay(ui: &nexus::imgui::Ui) {
+    let mut state = RenderState::lock();
+    state.draw(ui);
+    drop(state);
+}
+
+#[cfg(feature = "space")]
+fn render_space(ui: &nexus::imgui::Ui) {
+    if let Some(settings) = SETTINGS.get().and_then(|settings| settings.try_read().ok()) {
+        if settings.enable_katrender {
+            if !ENGINE_INITIALIZED.get() {
+                let (space_sender, space_receiver) = channel::<SpaceEvent>(32);
+                let _ = SPACE_SENDER.set(space_sender);
+                let drawstate_inner = Engine::initialise(ui, space_receiver);
+                if let Err(error) = &drawstate_inner {
+                    log::error!("DrawState setup failed: {error:?}");
+                };
+                ENGINE.set(drawstate_inner.ok());
+                ENGINE_INITIALIZED.set(true);
+            }
+            ENGINE.with_borrow_mut(|ds_op| {
+                if let Some(ds) = ds_op {
+                    if let Err(error) = ds.render(ui) {
+                        log::error!("Engine error: {error}");
+                    }
+                }
+            });
+        }
+    }
+}
+
+fn load_texture_bytes<K, B>(key: K, bytes: B) where
+    K: AsRef<str> + Into<String>,
+    B: AsRef<[u8]> + Into<Vec<u8>>,
+{
+    #[cfg(feature = "texture-loader")]
+    match rt::d3d11_device() {
+        Ok(Some(d3d11)) => {
+            match resources::Texture::new_bytes(&d3d11, bytes.as_ref(), key.as_ref()) {
+                Ok(texture) => {
+                    let mut gooey_lock = IMGUI_TEXTURES.get().unwrap().write().unwrap();
+                    if let Some(texture) = texture.to_nexus() {
+                        gooey_lock.entry(key.into())
+                            .or_insert(Arc::new(texture));
+                    }
+                    return
+                },
+                Err(e) => {
+                    log::warn!(target:"texture-loader", "failed to load {}: {e}", key.as_ref());
+                },
+            }
+        },
+        Err(e) => {
+            log::info!(target:"texture-loader", "D3D11 unavailable? {e}");
+        },
+        _ => (),
+    }
+
+    texture_schedule_bytes(key, bytes)
+}
+
+fn load_texture_path(rel: RelativePathBuf, path: PathBuf) {
+    // TODO: if load fails, mark it in hashmap to avoid repeately attempting load
+    // (regardless of load method, resources::texture or nexus or otherwise)
```

Comment:
this note is relevant uh, texture loading probably needs to be a little more stateful, maybe make a real struct to go into the `IMGUI_TEXTURES` map to track the distinction between loading vs ready vs missing/failed etc.